java_training/src/com/lftechnology/jan12/WordListManipulator.java

```diff
@@ -0,0 +1,132 @@
+package com.lftechnology.jan12;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class makes use of the Collection API and has methods to
+ * <ul>
+ * <li>find the duplicates in a word list</li>
+ * <li>find the unique words in a word list</li>
+ * <li>count the number of duplicates</li>
+ * <li>display the word list with duplicates removed</li>
+ * <li>to makes use of the Shuffle algorithm</li>
+ * </ul>
+ * 
+ * @author nerisa
+ * 
+ */
+public class WordListManipulator {
+	private static final Logger LOG = Logger.getLogger(WordListManipulator.class.getName());
+
+	/**
+	 * @author nerisa
+	 * @param listOfWords
+	 *            {@link List} of words from where the duplicates are to be removed
+	 * @return {@link List} of words with duplicates removed
+	 */
+	public Set<String> removeDuplicates(List<String> listOfWords) {
+		return (new HashSet<String>(listOfWords));
+	}
+
+	/**
+	 * @author nerisa
+	 * @param listOfWords
+	 *            {@link List} of words from where the duplicates are to be found
+	 * @return {@link List} of duplicates words
+	 */
+	public Set<String> findDuplicates(List<String> listOfWords) {
+		Set<String> noDuplicates = removeDuplicates(listOfWords);
+		Set<String> duplicates = new HashSet<String>();
+		for (String word : noDuplicates) {
+			if (Collections.frequency(listOfWords, word) > 1) {
+				duplicates.add(word);
+
+			}
+		}
+		return duplicates;
+	}
+
+	/**
+	 * @author nerisa
+	 * @param listOfWords
+	 *            {@link List} of words from where the unique words are to be found
+	 * @return {@link List} of unique words
+	 */
+	public Set<String> findUniques(List<String> listOfWords) {
+		Set<String> noDuplicates = removeDuplicates(listOfWords);
+		noDuplicates.removeAll(findDuplicates(listOfWords));
+		return (noDuplicates);
+	}
+
+	/**
+	 * Finds the number of times the duplicate words is repeated.
+	 * 
+	 * @author nerisa
+	 * @param listOfWords
+	 *            {@link List} of words form where the duplicates are to be found
+	 * @return returns the duplicates with its repetition count in the form of {@link Map}
+	 */
+	public Map<String, Integer> findRepetitionCount(List<String> listOfWords) {
+		Map<String, Integer> duplicates = new HashMap<String, Integer>();
+		Set<String> duplicateList = findDuplicates(listOfWords);
+		for (String word : duplicateList) {
+			duplicates.put(word, Collections.frequency(listOfWords, word));
+		}
+		return duplicates;
+	}
+
+	/**
+	 * Displays the duplicates with its repetition count
+	 * 
+	 * @author nerisa
+	 * @param map
+	 *            {@link Map} that contains the duplicates with its repetition count
+	 */
+	public void displayRepetitionCount(Map<String, Integer> map) {
+		Map<String, Integer> reversedMap = new TreeMap<String, Integer>(map);
```

Comment:
I guess you are trying to print *sorted Map*, so you used `TreeMap`, that's fine with me and :+1: .
But why would you need these block of statements when you have done this `new TreeMap<String, Integer>(map);`.

```
for (Map.Entry<String, Integer> entry : map.entrySet()) {
  reversedMap.put(entry.getKey(), entry.getValue());
}
```

What does this does?

Put same content from same variable to same variable in exact order of the previous value. or whatever!! It is not necessary.